content/blog/2022/12/2022-12-14-require-java-11.adoc

```diff
@@ -0,0 +1,239 @@
+about a year ahead of our usual timeline.
+This unusually aggressive timeline does not represent a change in policy for the Jenkins project
+but is rather an exception to the rule to facilitate migration to a number of breaking changes in the upstream Java community.
+
+Such a transition constitutes a https://en.wikipedia.org/wiki/Flag_day_(computing)[flag day], or breaking change.
+Below are answers to frequently asked questions,
+inspired by Uma Chingunde's excellent framework for https://umach.medium.com/communicating-organizational-changes-6277e4b47cff[communicating organizational changes] and building on the excellent https://community.jenkins.io/t/solving-failing-parent-pom-upgrade-from-4-51-to-4-52-cheatsheet/4958[cheat sheet] started by Jean-Marc Meessen.
+
+=== What is the change?
+
+Beginning with https://github.com/jenkinsci/plugin-pom/releases/tag/plugin-4.52[4.52],
+the plugin build toolchain requires Java 11 or newer and Jenkins 2.361 or newer.
+
+=== When is the change effective?
+
+The change is effective as of plugin parent POM 4.52, which was released on December 1, 2022.
+
+=== Why is it happening?
+
+Due to a large number of breaking changes in the upstream Java community,
+it became impractical to support both Java 8 and Java 11 in the same build toolchain.
+For more details, see https://groups.google.com/g/jenkinsci-dev/c/pjfvsMw-EMM/m/OIw_hveUBwAJ[the developer mailing list].
+
+=== Who is affected by the change?
+
+This change affects all plugin developers,
+particularly those who receive updates to the plugin build toolchain via Dependabot pull requests.
+
+=== What action do I need to take?
+
+At a high level, three actions need to be taken, the third of which depends on the first two:
+
+. Adjust the `Jenkinsfile` to use Java 11 or newer, removing any Java 8 configuration(s).
+. Update the baseline to 2.361 or newer.
+. Update the plugin parent POM to 4.52 or newer.
+
+We will elaborate on each of these three points below.
+
+==== Adjust the `Jenkinsfile` to use Java 11 or newer
+
+First, consult the matrix of build configurations in the plugin's `Jenkinsfile`.
+The goal is to ensure the plugin is building on Java 11 and 17 and not on Java 8,
+as recommended in https://github.com/jenkinsci/archetypes/blob/master/common-files/Jenkinsfile[the latest version of the archetype]:
+
+[source,groovy]
+----
+buildPlugin(
+  useContainerAgent: true, // Set to `false` if you need to use Docker for containerized tests
+  configurations: [
+    [platform: 'linux', jdk: 17],
+    [platform: 'windows', jdk: 11],
+])
+----
+
+Note the explicit `configurations` entry with two versions: Java 11 and 17.
+If there is an explicit configuration for Java 8, remove it -- Java 8 is no longer supported as of plugin parent POM 4.52.
+
+NOTE: If you are not already using container agents, we recommend adding `useContainerAgent: true` (but this is not mandatory).
```

Comment:
```suggestion
NOTE: If you are not already using container agents, we recommend adding `useContainerAgent: true` (but this is not mandatory).

Doing so result in ci.jenkins.io spawning a link:https://github.com/jenkins-infra/documentation/blob/main/ci.adoc#container-agents[Container agent] for executing the builds instead of of a Virtual Machine: usually faster to start and generates less costs for the project.
```

Reply:
As with the majority of ostensible suggestions I receive, the intention behind this suggestion is valid, but the suggested implementation is actually a non-suggestion, in this case because it contains numerous AsciiDoc formatting errors and grammatical errors. Commit 84fb58e9d00f6fb5ce5dc2bc28a4e8c6634e5b0e demonstrates how to turn this non-suggestion into a valid suggestion by implementing it properly through AsciiDoc block admonitions, proper capitalization, etc.